tests/unit/inference/test_inference.py

```diff
@@ -544,6 +544,39 @@ def test(
         print(local_rank, "deepspeed", ds_output)
         assert assert_fn(bs_output, ds_output)
 
+    @pytest.mark.world_size(3)
+    def test_odd_world_size(
+        self,
+        model_w_task,
+        query,
+        inf_kwargs,
+        assert_fn,
+        dtype,
+    ):
+        invalid_test_msg = validate_test(model_w_task, dtype, enable_cuda_graph=False, enable_triton=False)
+        if invalid_test_msg:
+            pytest.skip(invalid_test_msg)
+
+        model, task = model_w_task
+        if model == "Salesforce/codegen-350M-mono":
+            pytest.skip("codegen does not supported by odd world_size")
+        local_rank = int(os.getenv("LOCAL_RANK", "0"))
+        world_size = int(os.getenv("WORLD_SIZE", "3"))
+
+        # We have to load these large models on CPU with pipeline because not
+        # enough GPU memory
+        pipe = pipeline(task, model=model, device=torch.device("cpu"), framework="pt")
```

Comment:
The models we are testing here are not too large to fit on GPU. I think this was copied from another test when `TestAutoTensorParallelism` was created - but this is wrong. If you can, please also reflect this change in `TestAutoTensorParallelism.test`. Thank you!
```suggestion
        pipe = pipeline(task, model=model, device=torch.device(get_accelerator().device_name(local_rank)), framework="pt")
```

Reply:
These two changes had been made in latest update.  Thanks!